Frontend/nutri-pal/src/app/gates/register/page.tsx

```diff
@@ -17,38 +20,73 @@ export default function register() {
+            setPasswordSame(false);
+        } else {
+            setPasswordSame(true);
+        }
+
+    }, [confirmPassword])
+
     return (
         <main className="register-main-container">
             <div className='register-form-container'>
                 <div className='register-text'>
                     <button onClick={onBackArrow}>
                         <FaArrowCircleLeft size={ARROW_SIZE} className='register-back-icon' />
                     </button>
-                    <p>Create account</p>
+                    <p className='register-title'>Create account</p>
                 </div>
                 <form className='register-form' onSubmit={onSubmitForm}>
                     <div className='register-name-last-container'>
                         <div className='register-name-container'>
-                            <Inputs id='name' label='Name' placeholder='Name' Icon={FaUser} type='text' setValue={setName} />
+                            <Inputs id='name' label='Name' placeholder='Name' Icon={FaUser} type='text' setValue={setName} required={true} />
                         </div>
                         <div className='register-last-container'>
-                            <Inputs id='lastname' label='Last Name' placeholder='Last Name' Icon={FaUser} type='text' setValue={setLastName} />
+                            <Inputs id='lastname' label='Last Name' placeholder='Last Name' Icon={FaUser} type='text' setValue={setLastName} required={true} />
                         </div>
                     </div>
-                    <Inputs id='username' label='Username' placeholder='Username' Icon={FaUser} type='text' setValue={setUsername} />
-                    <Inputs id='email' label='Email' placeholder='email@example.com' Icon={FaUser} type='text' setValue={setEmail} />
-                    <Inputs id='password' label='Password' placeholder='Password' Icon={FaLock} type='password' setValue={setPassword} />
-                    <Inputs id='confirm' label='Confirm Password' placeholder='Password' Icon={FaLock} type='password' setValue={setConfirmPassword} />
+                    <Inputs id='username' label='Username' placeholder='Username' Icon={FaUser} type='text' setValue={setUsername} required={true} />
+                    <Inputs id='email' label='Email' placeholder='email@example.com' Icon={FaUser} type='text' setValue={setEmail} required={true} />
+                    <Inputs id='password' label='Password' placeholder='Password' Icon={FaLock} type='password' setValue={setPassword} required={true} />
+                    <div>
+                        <Inputs id='confirm' label='Confirm Password' placeholder='Password' Icon={FaLock} type='password' setValue={setConfirmPassword} required={true} passwordSame={passwordSame} />
+                        {(!passwordSame) ? <p className='register-confirm-password-error'>Please make sure your passwords match</p> : null}
```

Comment:
instead of using state, why not just do:

confirmPassword === password && confirmPassword !== ''